### ISSUE TITLE
Feature Request: Per Function Statistics

### ISSUE THREAD

**Issue by richi235** [dspinellis/cqmetrics#1]
I'm currently writing a research work including code quality metrics for linux tcp and lwip tcp.
For this I want per-function metrics. I wrote a small perl script which split the functions appart
and fed them to cqmetrics seperately. 

But for future uses such a cqmetrics built in functionality would be nice.

**Comment by dspinellis** [comment]
So cqmetrics should output a separate line for each function?

**Comment by richi235** [comment]
Yes, this is what I thought, and what my wrapper script finally outputed

**Comment by dspinellis** [comment]
Probably better to implement this as block, where functions are the top-level block for C, while classes can be the top-level block for C++ and Java.

**Comment by dspinellis** [comment]
Which specific metrics did you find useful on a per-function level?
